samcli/local/docker/container.py

```diff
@@ -398,21 +404,21 @@ def wait_for_result(self, full_path, event, stdout, stderr, start_timer=None):
         if timer:
             timer.cancel()
 
-        # NOTE(jfuss): Adding a sleep after we get a response from the contianer but before we
-        # we write the response to ensure the last thing written to stdout is the container response
-        time.sleep(1)
+        self._logs_thread_event.wait()
```

Comment:
Does `.wait` has timeout option? If so we might want to set it in case it keeps waiting indefinetely when the message below never gets printed (I guess only possible if RIE/rapid crashes).